### PR TITLE
Shell correctness improvements

### PR DESCRIPTION
**Pull request by CyberShadow**
Hi, and thank you for publishing this project!

Considering that this project deals with matters related to privacy and personal security, I think it's important that the produced results are as expected. With this in mind, I would like to contribute the following set of changes, which attempt to address a small number of correctness-related practical issues and a larger number of pedantic issues. Please see the commit messages for details, and I'll be happy to further elaborate on any questions.

### CODE REVIEW

**Comment by bananer** [comment]
Seems like those are good suggestions, thank you very much!

I need to look into it a bit more carefully before merging though.

**Review by bananer — APPROVED** [review]

**Comment by bananer** [comment]
Thank you for your contributions, @CyberShadow !

File: src/build.sh [hunk 1]
```diff
@@ -311,15 +312,15 @@ for branch in ${BRANCH_NAME//,/ }; do
         # Start the build
         echo ">> [$(date)] Starting build for $codename, $branch branch" | tee -a "$DEBUG_LOG"
         build_successful=false
-        if brunch "$codename" &>> "$DEBUG_LOG"; then
+        if ( set +eu ; brunch "$codename" ) &>> "$DEBUG_LOG"; then
```
Comment (bananer) [inline src/build.sh:315]: Wait, are we forgetting to set `-eu` back after that command? @CyberShadow

Comment (CyberShadow) [reply src/build.sh:315]: No; it is done in a subshell, so the setting is scoped to it.

Comment (bananer) [reply src/build.sh:315]: Alright, thanks for clarifying!

**Comment by bananer** [comment]
Unfortunately this has caused (some?) builds to fail and I needed to revert. See #253 

I did not investigate much but it seems like some parts of the android/lineage build system don't like the changed shell options.

Maybe we can carefully apply some of the other changes still, do it one shellcheck rule at a time and with proper testing.

**Comment by CyberShadow** [comment]
I'm sorry, but I think bluntly reverting this and #174 was a bad call. Doing so reintroduced correctness and (depending on deployment) potential security issues.

At the time when I submitted the pull request (five months ago), the build was succeeding. I would suggest scrutinizing changes that have been made since this pull request was opened instead.

> I did not investigate much but it seems like some parts of the android/lineage build system don't like the changed shell options.

If recent changes introduced calls to functions defined in sourced Android shell libraries, then these ought to be updated to be invoked with the flags turned off, as in the change discussed above.

**Comment by CyberShadow** [comment]
@bananer I have not tested it, but this merge resolution might not be correct:

https://github.com/lineageos4microg/docker-lineage-cicd/commit/d99bcbd3f3692c970db22274e6a300ef444323c9#diff-02f64ab7fe6df7a34979c489dccf70a9902438ab964f4fe9471534d6b7b74a76L306-R307

If `breakfast` sets any shell variables which are needed by `mka bacon`, they will not be visible once the subshell exits. In this case, the correct way to invoke it is to temporarily unset `eu`, then set it back on, as done when sourcing `build/envsetup.sh` in this pull request.

**Comment by bananer** [comment]
My priority was to restore a working state and reverting your PR was the first thing I tried.

Thanks for looking into this, I agree my merge resolution between this and #252 likely introduced the problem.

**Pull request merged**